### PR TITLE
Optimize Document Post-it perspective LMR-329

### PR DESCRIPTION
**Pull request by justmesr**

### CODE REVIEW

File: src/main/scripts/src/app/core/dto/query.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 export interface Query {
 
-  collectionCodes?: string[];
+  collections?: string[];
```
Comment (livthomas) [inline src/main/scripts/src/app/core/dto/query.ts]: Why have you changed this? It follows the structure of DTO objects defined the back-end. See [JsonQuery](https://github.com/Lumeer/engine/blob/devel/api/src/main/java/io/lumeer/api/dto/JsonQuery.java). Does the search work after this change?

File: src/main/scripts/src/app/core/header/header.component.html [hunk 2]
```diff
@@ -11,18 +11,21 @@
 
 <div class="menu">
   <div class="project" *ngIf="workspaceService.isWorkspaceSet()">
-    <a [routerLink]="['/workspace']">
+    <a tabindex [routerLink]="['/workspace']">
```
Comment (livthomas) [inline src/main/scripts/src/app/core/header/header.component.html]: Are these changes really needed? I think the TAB button works well even without them.

File: src/main/scripts/src/app/core/rest/collection.service.ts [hunk 3]
```diff
@@ -113,9 +113,20 @@ export class CollectionService {
       .catch(CollectionService.handleGlobalError);
   }
 
+  private toDto(collection: Collection): Collection {
```
Comment (livthomas) [inline src/main/scripts/src/app/core/rest/collection.service.ts]: Is this method really needed? It just creates the object with the same structure as the already existing one. Or where is the catch?

File: src/main/scripts/src/app/shared/permissions/table/permissions-table.component.scss [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 
 th {
-  border: 1px solid #eeeeee;
+  border: 1px solid $grey-color;
   padding: 10px 20px;
 }
 
 td {
-  border: 1px solid #eeeeee;
+  border: 1px solid $grey-color;
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/permissions/table/permissions-table.component.scss]: We should probably use only Boostrap classes for styling tables so these styles will not be needed anymore.

File: src/main/scripts/src/app/shared/perspectives/documents/post-it/document/post-it-document.component.ts [hunk 5]
```diff
@@ -104,38 +66,16 @@ export class PostItDocumentComponent implements OnInit {
   }
 
   private setEventListener(): void {
-    // disable scrolling when navigating using keys
+    // disable scrolling when navigating using up and down keys
     this.content.nativeElement.addEventListener('keydown', (key: KeyboardEvent) => {
-      [37, 38, 39, 40].includes(key.keyCode) && key.preventDefault();
+      [38, 40].includes(key.keyCode) && key.preventDefault();
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/documents/post-it/document/post-it-document.component.ts]: Isn't it better to use keycode constants? I think they are defined somewhere. It is not obvious now which keys are used here. We should avoid comments in code and make it self-describing instead.

File: src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon/icon-palette.component.ts [hunk 6]
```diff
@@ -33,7 +33,7 @@ export class IconPaletteComponent extends Palette {
   @Input('icon')
   public active: string;
 
-  public icons = Const.icons.slice(0, 100);
+  public icons = Const.icons.slice(0, 10);
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon/icon-palette.component.ts]: Just ten icons?

File: src/main/scripts/src/app/core/dto/query.ts (hunk 1 above)
Comment (justmesr) [reply src/main/scripts/src/app/core/dto/query.ts]: I guess it doesn't if it did before.. just noticed that in constructor it takes `collections` (https://github.com/Lumeer/engine/blob/devel/api/src/main/java/io/lumeer/api/dto/JsonQuery.java#L73) so I changed it to that.

File: src/main/scripts/src/app/core/header/header.component.html (hunk 2 above)
Comment (justmesr) [reply src/main/scripts/src/app/core/header/header.component.html]: `<a>` elements without `tabindex` or `href` in new Bootstrap change color to that of regular text on hover.. so instead of messing with Bootstrap styles I rather added it.

File: src/main/scripts/src/app/core/rest/collection.service.ts (hunk 3 above)
Comment (justmesr) [reply src/main/scripts/src/app/core/rest/collection.service.ts]: The collection it receives from backend has additional attributes (like the `lastTimeUsed`) that caused the update method to fail.

File: src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon/icon-palette.component.ts (hunk 6 above)
Comment (justmesr) [reply src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon/icon-palette.component.ts]: Just to fix the slow loading (And they don't work after the FA update anyway).. I am changing the icon picker a bit in a PR with the rest of Martin's requested changes.
